rust/cubestore-sql-tests/src/tests.rs

```diff
@@ -2660,6 +2661,40 @@ async fn float_index(service: Box<dyn SqlClient>) {
     }
 }
 
+async fn date_add(service: Box<dyn SqlClient>) {
+    let r = service
+        .exec_query(
+            "SELECT
+            DATE_ADD(CAST('2021-01-01T00:00:00Z' as TIMESTAMP), INTERVAL '1 second'),\
```

Comment:
Could you add a few more tests?
At least these corner cases should be covered:
- Add one day to Jun 30
- Add one month to Dec
- Add one day to Feb 28 on a leap year and non-leap year
- add one year to Feb 29 on a leap year
- add one minute to 23:59:00
- add one month to Jan 30
- add one month to Jan 29 on a non-leap year and a leap year


Is there a way to get negative values in the intervals?
If no, please add assertions.
If yes, please add tests.